Sources/SocketForwarder/LRUCache.swift

```diff
@@ -0,0 +1,107 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the container project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+struct KeyExistsError: Error {}
+
+class LRUCache<K: Hashable, V> {
+    private class Node {
+        fileprivate var prev: Node?
+        fileprivate var next: Node?
+        fileprivate let key: K
+        fileprivate let value: V
+
+        init(key: K, value: V) {
+            self.prev = nil
+            self.next = nil
+            self.key = key
+            self.value = value
+        }
+    }
+
+    private var head: Node?
+    private var tail: Node?
+    private var members: [K: Node]
+
+    public init() {
+        self.head = nil
+        self.tail = nil
+        self.members = [:]
+    }
+
+    public var count: Int { members.count }
+
+    public func get(_ key: K) -> V? {
+        guard let node = members[key] else {
+            return nil
+        }
+        listRemove(node: node)
+        listInsert(node: node, after: tail)
+        return node.value
+    }
+
+    public func put(key: K, value: V) throws {
+        if members[key] != nil {
+            throw KeyExistsError()
+        }
+
+        let node = Node(key: key, value: value)
+        listInsert(node: node, after: tail)
+        members[key] = node
+    }
+
+    public func evict() -> (K, V)? {
```

Comment:
I'm surprised to see this, I'd think the constructor would take in some bounds (which for this is the maximum number of proxies we allow) and then the evictions would happen internally in the object. I see we use to so that we can get a handle to what is being evicted so we can call close on it, but couldn't the type we're adding just have a destructor that calls close internally? It didn't seem like `ProxyContext.context.close()` throws so this should function

Reply:
The reason this is somewhat odd to me is the eviction is handled externally by the user of this structure which doesn't seem common, although this is a tailored type for our needs so it's not the biggest concern.

Reply:
When I was writing this I thought I'd put the minimal amount of stuff in the cache and leave the sizing entirely to the client. I expressly didn't want to couple the cache with anything special about the type; that is to say I didn't want to make it a cache of `V: Evictable` where that protocol defined some `evictYourself()`.

Perhaps it's better called a LRUTracker since it's up to the client to bound the size, whereas LRUCache evokes a cache with a predefined maximum number of entries?

I'm fine giving it fixed size behavior. I think if we had `LRUCache(size: UInt)` and changed put to:

```swift
func put(key: K, value: V) -> V?
```

maybe that makes the semantics more clear?

Reply:
I'll push the other changes now and then knock out that change.